### PR TITLE
feat(spel): New `currentStage()` function

### PR DESCRIPTION
**Pull request by ajordens**
This fn provides a direct reference to the currently executing `Stage`.

### CODE REVIEW

**Comment by ajordens** [comment]
Couple tests forthcoming but wanted to get the PR going before laptop went from 7% -> 0%.

File: orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/expressions/ExpressionTransform.java [hunk 1]
```diff
@@ -266,7 +272,18 @@ static String escapeSimpleExpression(String expression) {
    */
   private String includeExecutionParameter(String e) {
     String expression = e;
-    for (String fn : EXECUTION_AWARE_FUNCTIONS) {
+
+    // An expression aware function is any that takes an Execution as its first parameter
+    Set<String> expressionAwareFunctions = new HashSet<>(EXECUTION_AWARE_FUNCTIONS);
```
Comment (ajordens) [inline orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/expressions/ExpressionTransform.java]: Wouldn't mind splitting the other stage-related functions out of `ExpressionSupport` and out of the hardcoded `EXECUTION_AWARE_FUNCTIONS` constant ... was thinking that would be a subsequent PR (I don't _think_ anything should break!).

File: orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/expressions/functions/StageExpressionFunctionProvider.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            )
+        ))
+    );
+  }
+
+  /**
+   * @param execution the current execution
+   * @return the currently executing stage
+   */
+  public static Stage currentStage(Execution execution) {
```
Comment (ajordens) [inline orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/expressions/functions/StageExpressionFunctionProvider.java]: I'm not sure the history of the other functions taking an `Object` vs `Execution` explicitly ... this worked in my testing.

Comment (robzienert) [reply orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/expressions/functions/StageExpressionFunctionProvider.java]: Cool. I wasn't sure what the reason was for not using `Execution` either.

**Pull request merged**